p2panda-rs/src/keypair.rs

```diff
@@ -73,16 +76,25 @@ impl KeyPair {
         Box::from(self.0.secret.to_bytes())
     }
 
-    /// Sign data using this key pair.
-    #[cfg(not(target_arch = "wasm32"))]
-    pub fn sign(&self, data: &[u8]) -> Signature {
-        self.0.sign(data)
+    /// Sign a message using this key pair.
+    pub fn sign(&self, message: &[u8]) -> Box<[u8]> {
```

Comment:
Do we want the `Box` here? I'm not sure I fully understand the consequences of that but it works and that made me happy :D

Reply:
I can recommend checking the wasm-bindgen documentation, there you have an overview of allowed return types (coming from Rust -> JS or vice-versa): https://rustwasm.github.io/docs/wasm-bindgen/reference/types/boxed-jsvalue-slice.html - so yeah, looks good :+1:

Reply:
Okay I was really vague, I'm sorry :D I had looked at those docs, what I meant was more like: Do we introduce a significant performance penalty by `Box`ing the return value of a function that will be called *a lot*? We are only doing it here for the Javascript side of things so if there is a disadvantage we could have separate implementations for js and rs world with only the js one returning a Box value. Now that I'm writing it down I realise that we are not at a point in time where this matters :D